cmd/pitr/pxc/pxc.go

```diff
@@ -187,12 +208,80 @@ func GetPXCLastHost(pxcServiceName string) (string, error) {
 		}
 	}
 	if len(lastHost) == 0 {
-		return "", errors.New("cant find host")
+		return "", errors.New("can't find host")
 	}
 
 	return lastHost, nil
 }
 
+func GetPXCOldestBinlogHost(pxcServiceName, user, pass string) (string, error) {
+	cmd := exec.Command("peer-list", "-on-start=/usr/bin/get-pxc-state", "-service="+pxcServiceName)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrap(err, "get peer-list output")
+	}
+	nodes := strings.Split(string(out), "node:")
+
+	var oldestHost string
+	var oldestTS int64
+	for _, node := range nodes {
+		if strings.Contains(node, "wsrep_ready:ON:wsrep_connected:ON:wsrep_local_state_comment:Synced:wsrep_cluster_status:Primary") {
+			nodeArr := strings.Split(node, ":")
+			db, err := NewPXC(nodeArr[0], user, pass)
```

Comment:
we have a lot of continue there
to make it more `clear` i think it's worth to put `db.Close()` to defer after opening connection

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/761/commits/f88ca5e04a895030dc7ef74f1397cecb900c20a1